Packs/SysAid/Integrations/SysAid/SysAid.py

```diff
@@ -784,19 +881,95 @@ def service_record_delete_command(client: Client, args: Dict[str, Any]) -> Comma
+    else:
+        msg = f'Error {response.status_code} occurred while uploading file to service record {id}.'
+
+    command_results = CommandResults(
+        outputs_prefix='SysAid.ServiceRecord',
+        readable_output=msg
+    )
+
+    return command_results
+
+
+def service_record_delete_file_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    sr_id = str(args.get('id'))
+    file_id = str(args.get('file_id'))
+
+    response = client.service_record_delete_file_request(sr_id, file_id)
+    if response.status_code == 200:
+        msg = f'File deleted from Service Record {sr_id} successfully.'
+    else:
+        msg = f'Error {response.status_code} occurred while deleting file from service record {id}.'
+
+    command_results = CommandResults(
+        outputs_prefix='SysAid.ServiceRecord',
+        readable_output=msg
+    )
+
+    return command_results
+
+
+def service_record_get_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    sr_id = str(args.get('id'))
+    fields = set_returned_fields(args.get('fields'))
+
+    response = client.service_record_get_request(sr_id, fields)
+    headers = ['id', 'title', 'Status', 'Modify time', 'Service Record Type', 'notes']
+    readable_response = create_readable_response(response, service_record_handler)
+    command_results = CommandResults(
+        outputs_prefix='SysAid.ServiceRecord',
+        outputs_key_field='id',
+        outputs=response,
+        raw_response=response,
+        readable_output=tableToMarkdown('Service Record Results:',
+                                        readable_response,
+                                        headers=headers,
+                                        removeNull=True,
+                                        headerTransform=pascalToSpace)
+    )
+
+    return command_results
+
+
+def service_record_add_note_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    sr_id = str(args.get('id'))
+    note = args.get('note')
+    username = args.get('username')
+
+    client.service_record_add_note_request(sr_id, note, username)
+    msg = 'Updated record with new note'
+
+    command_results = CommandResults(
+        outputs_prefix='SysAid.ServiceRecord',
+        readable_output=msg
+    )
+
+    return command_results
+
+
 def fetch_incidents(client: Client, first_fetch: str, limit: Optional[int] = MAX_INCIDENTS_TO_FETCH,
                     included_statuses: str = None, include_archived: bool = False, fetch_types: str = None):
     last_fetch = demisto.getLastRun().get('last_fetch')
     last_id_fetched = demisto.getLastRun().get('last_id_fetched', '-1')
     fetch_start_datetime = calculate_fetch_start_datetime(last_fetch, first_fetch)
     demisto.debug(f'last fetch was at: {last_fetch}, last id fetched was: {last_id_fetched}, '
                   f'time to fetch from is: {fetch_start_datetime}.')
+    filter_times = None
```

Comment:
Change it to hold the wanted value

Reply:
This is the wanted value.

Reply:
When will it not be `None` and will be used in `fetch_request` as the `insert_time`?
This is the main change we talked about, I think it must hold a meaningful value.
consider using `fetch_start_datetime` in order to set this value.